src/foam/u2/view/FilteredReferenceView.js

```diff
@@ -0,0 +1,93 @@
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'FilteredReferenceView',
+  extends: 'foam.u2.View',
+
+  requires: [
+    'foam.u2.ReferenceView',
+  ],
+
+  implements: [
+    'foam.mlang.Expressions'
+  ],
+
+  exports: [
+    'filteredDAO'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'firstDAOKey',
+      documentation: 'The key of the DAO that the user makes the second selection from. This DAO will be filtered based on the first selection.'
```

Comment:
I think you might have the comments backwards. This would be the key of the _first_ DAO that they make a selection from, right? Then the second DAO is filtered based on the selection from the first DAO.